modules/sov-ibc/src/rpc/context.rs

```diff
@@ -1,30 +1,345 @@
+use borsh::BorshSerialize;
 use ibc_client_tendermint::types::client_type as tm_client_type;
-use ibc_core::channel::types::channel::IdentifiedChannelEnd;
+use ibc_core::channel::types::channel::{ChannelEnd, IdentifiedChannelEnd};
+use ibc_core::channel::types::commitment::{AcknowledgementCommitment, PacketCommitment};
 use ibc_core::channel::types::error::ChannelError;
-use ibc_core::channel::types::packet::PacketState;
+use ibc_core::channel::types::packet::{PacketState, Receipt};
 use ibc_core::client::context::ClientValidationContext;
 use ibc_core::client::types::Height;
 use ibc_core::connection::types::error::ConnectionError;
-use ibc_core::connection::types::IdentifiedConnectionEnd;
+use ibc_core::connection::types::{ConnectionEnd, IdentifiedConnectionEnd};
 use ibc_core::handler::types::error::ContextError;
 use ibc_core::host::types::identifiers::{ChannelId, ClientId, ConnectionId, PortId, Sequence};
 use ibc_core::host::types::path::{
-    AckPath, ChannelEndPath, ClientConnectionPath, ClientConsensusStatePath, CommitmentPath, Path,
-    ReceiptPath,
+    AckPath, ChannelEndPath, ClientConnectionPath, ClientConsensusStatePath, CommitmentPath,
+    ConnectionPath, Path, ReceiptPath, SeqAckPath, SeqRecvPath, SeqSendPath, UpgradeClientPath,
 };
 use ibc_core::host::{ClientStateRef, ConsensusStateRef, ValidationContext};
 use ibc_query::core::context::{ProvableContext, QueryContext};
+use sov_celestia_client::client_state::ClientState as HostClientState;
+use sov_celestia_client::consensus_state::ConsensusState as HostConsensusState;
 use sov_modules_api::Spec;
 
 use crate::context::IbcContext;
+use crate::helpers::StorageValue;
+
+impl<'a, S> IbcContext<'a, S>
+where
+    S: Spec,
+{
+    pub fn dyn_client_state<SV>(&self, client_id: &ClientId) -> SV::Output
+    where
+        SV: StorageValue<ClientStateRef<Self>>,
+    {
+        SV::value_at_key(
+            client_id,
+            &self.ibc.client_state_map,
+            *self.working_set.borrow_mut(),
+        )
+    }
+
+    pub fn dyn_client_consensus_state<SV>(
```

Comment:
I am more inclined to name these methods as `query_*` instead of `dyn_*`.

Reply:
a1b7e28